src/lib.rs

```diff
@@ -62,7 +77,16 @@ fn process_includes(
 
     while let Some(captures) = INCLUDE_RE.captures(&result.clone()) {
         let capture = captures.get(0).unwrap();
-        let include_path = resolve_path(captures.name("file").unwrap().as_str());
+
+        let include_path;
+        #[cfg(feature = "relative-path")] {
+            let mut source_dir = source_path.to_path_buf();
+            source_dir.pop();
+            include_path = resolve_path(captures.name("file").unwrap().as_str(), Some(&source_dir));
+        }
+        #[cfg(not(feature = "relative-path"))] {
+            include_path = resolve_path(captures.name("file").unwrap().as_str(), None);
+        }
```

Comment:
In order to reduce code complexity and feature flag checks, could we please refacto this part?

```suggestion
        #[allow(unused_assignments)]
        let mut include_parent_dir_path = None;
        
        #[cfg(feature = "relative-path")] {
            let mut path = source_path.to_path_buf();
            path.pop();
            include_parent_dir_path = Some(path);
        }
        
        let include_path = resolve_path(captures.name("file").unwrap().as_str(), include_parent_dir_path);
```